### PR TITLE
Refactor gpu test to use RunIf

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

While debugging some gpu tests, came across this test.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/trainer/test_trainer.py [hunk 1]
```diff
@@ -1096,17 +1096,13 @@ def test_invalid_gradient_clip_algo(tmpdir):
...
+@RunIf(min_gpus=1)
 def test_gpu_choice(tmpdir):
-    trainer_options = dict(default_root_dir=tmpdir)
-    # Only run if CUDA is available
-    if not torch.cuda.is_available():
-        return
-
     num_gpus = torch.cuda.device_count()
-    Trainer(**trainer_options, accelerator="gpu", devices=num_gpus, auto_select_gpus=True)
+    Trainer(default_root_dir=tmpdir, accelerator="gpu", devices=num_gpus, auto_select_gpus=True)
```
Comment (rohitgr7) [inline tests/trainer/test_trainer.py:1102]: tmpdir isn't required, but fine anyways

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: tests/trainer/test_trainer.py (hunk 1 above)
Comment (kaushikb11) [reply tests/trainer/test_trainer.py]: yup, updated!

**Comment by Sunset17** [comment]
Is this because of something I did? 


Sent from Yahoo Mail for iPhone


On Monday, April 4, 2022, 09:35, Kaushik B ***@***.***> wrote:




@kaushikb11 commented on this pull request.

In tests/trainer/test_trainer.py:
>      num_gpus = torch.cuda.device_count()
-    Trainer(**trainer_options, accelerator="gpu", devices=num_gpus, auto_select_gpus=True)
+    Trainer(default_root_dir=tmpdir, accelerator="gpu", devices=num_gpus, auto_select_gpus=True)

yup, updated!
